### PR TITLE
db-debugger

### PR DESCRIPTION
**Pull request by msmouse**
### Description

Creating db-debugger CLI to examine the DB and take DB checkpoints.
The tool needs to be part of the aptos-db crate to be able to access the internals, and have to be gated behind a feature since it introduces additional dependencies (clap, owo-colors)

image

image

image

Also see https://github.com/aptos-labs/aptos-core/pull/1465 where it was originally created.

(creating checkpoint doesn't work while DB is opened elsewhere already) 

### Test Plan
Played locally.

### CODE REVIEW

File: storage/aptosdb/src/db_debugger/checkpoint/mod.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    db_dir: DbDir,
+
+    #[clap(long, parse(from_os_str))]
+    output_dir: PathBuf,
+}
+
+impl Cmd {
+    pub fn run(self) -> Result<()> {
+        ensure!(!self.output_dir.exists(), "Output dir already exists.");
+        fs::create_dir_all(&self.output_dir)?;
```
Comment (grao1991) [inline storage/aptosdb/src/db_debugger/checkpoint/mod.rs:22]: Share the code with https://github.com/aptos-labs/aptos-core/blob/main/execution/executor-benchmark/src/lib.rs#L55 to make our life easier when adding more DBs / sharding DB?

**Review by grao1991 — APPROVED** [review]

File: storage/aptosdb/src/db_debugger/checkpoint/mod.rs (hunk 1 above)
Comment (msmouse) [reply storage/aptosdb/src/db_debugger/checkpoint/mod.rs:22]: done

**Review by zekun000 — APPROVED** [review]

File: storage/aptosdb/src/lib.rs [hunk 2]
```diff
@@ -695,17 +699,40 @@ impl AptosDB {
     }
 
     /// Creates new physical DB checkpoint in directory specified by `path`.
-    pub fn create_checkpoint<P: AsRef<Path>>(&self, path: P) -> Result<()> {
+    pub fn create_checkpoint(db_path: impl AsRef<Path>, cp_path: impl AsRef<Path>) -> Result<()> {
```
Comment (areshand) [inline storage/aptosdb/src/lib.rs:702]: update the comment above describing cp_path?

File: storage/aptosdb/src/db_debugger/ledger/check_rxn_info_hashes.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
```
Comment (areshand) [inline storage/aptosdb/src/db_debugger/ledger/check_rxn_info_hashes.rs:4]: should this file renamed to `check_txn_info_hashes.rs`

**Pull request merged**